src/clients/graphql/graphql_client.ts

```diff
@@ -26,7 +26,7 @@ export class GraphqlClient {
     this.client = new HttpClient(this.domain);
   }
 
-  async query(params: GraphqlParams): Promise<RequestReturn> {
+  async query<T = unknown>(params: GraphqlParams): Promise<RequestReturn<T>> {
```

Comment:
We can look into improving the default type here - right now this will map to `body: unknown` by default, so we could refactor that default later on to be more representative of an actual response.